#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name       Type      Level Description     Default Min   Max
gen.add("rotation_before_translation", bool_t, 0, "Base rotates before being translated", False)
gen.add("x_vel", double_t, 0, "X translation velocity", 0.1, 0.01, 0.5)
gen.add("y_vel", double_t, 0, "Y translation velocity", 0.1, 0.01, 0.5)
gen.add("theta_vel", double_t, 0, "Rotation velocity", 0.1, 0.01, 0.5)
gen.add("distance_tolerance", double_t, 0, "Distance tolerance", 0.01, 0.001, 0.01)

exit(gen.generate("mcr_relative_base_controller", "relative_base_controller_node", "RelativeBaseController"))
